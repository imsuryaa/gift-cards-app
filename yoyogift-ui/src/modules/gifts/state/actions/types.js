export const FETCH_CARDS = 'FETCH_CARDS';
export const FETCH_CARD = 'FETCH_CARD';
export const FETCH_CARD_FILTER = 'FETCH_CARD_FILTER';
export const ADMIN_ADD_CARD = 'ADMIN_ADD_CARD';
export const ADMIN_UPDATE_CARD = 'ADMIN_UPDATE_CARD';
export const DELETE_CARD = 'DELETE_CARD';
export const UPDATE_CARD_COUNT = 'UPDATE_CARD_COUNT';