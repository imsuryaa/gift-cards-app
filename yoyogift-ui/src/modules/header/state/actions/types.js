export const LOGIN = 'LOGIN';
export const LOGOUT = 'LOGOUT';
export const CREATE_USER= 'CREATE_USER';