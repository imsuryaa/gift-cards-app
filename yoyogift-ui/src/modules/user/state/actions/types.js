export const RECEIVED_CARDS = 'RECEIVED_CARDS';
export const SENT_CARDS = 'SENT_CARDS';
export const USER_DETAILS = 'USER_DETAILS';
export const REDEEM_CARD = 'REDEEM_CARD';
export const UPDATE_BALANCE = 'UPDATE_BALANCE';
export const UPDATE_TRANSACT = 'UPDATE_TRANSACT';