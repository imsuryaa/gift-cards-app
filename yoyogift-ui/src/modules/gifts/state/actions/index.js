import { FETCH_CARDS, FETCH_CARD, FETCH_CARD_FILTER, ADMIN_ADD_CARD, UPDATE_CARD_COUNT, ADMIN_UPDATE_CARD } from './types';
import axiosWrapper from '../../../../apis/axiosCreate';

export const fetchCards = () => async (dispatch) => {
    const response = await axiosWrapper.get('/giftCards');    
    dispatch ({
        type: FETCH_CARDS,
        payload: response
    })
}

export const fetchCard = (id) => async (dispatch) => {
    const response = await axiosWrapper.get(`/giftCards/${id}`);
    dispatch ({
        type: FETCH_CARD,
        payload: response
    })
}

export const adminUpdateCard = (id, object) => async (dispatch) => {
    const response = await axiosWrapper.patch(`/giftCards/${id}`, object);
    dispatch ({
        type: ADMIN_UPDATE_CARD,
        payload: response
    })
}

export const fetchCardFilter = (object) => async (dispatch) => {
    dispatch ({
        type: FETCH_CARD_FILTER,
        payload: object
    })
}

export const adminAddCard = (object) => async(dispatch) => {
    const response = await axiosWrapper.post('/giftCards', object);
    dispatch ({
        type: ADMIN_ADD_CARD,
        payload: response
    })
}

export const updateCardCount = (cardId, cardCount) => async(dispatch) => {
    const response = await axiosWrapper.patch(`/giftCards/${cardId}`, { cardCount: cardCount});
    dispatch ({
        type: UPDATE_CARD_COUNT,
        payload: response.data
    })
}

export const fetchCardsForMasonry = (rowLimit, pageNo) => async (dispatch) => {
    const response = await axiosWrapper.get('/giftCards',{
        params: {
          _limit: rowLimit,
          _page: pageNo
         }
      });    
    dispatch ({
        type: FETCH_CARDS,
        payload: {
            response,
            pageNo
        }
    })
}
