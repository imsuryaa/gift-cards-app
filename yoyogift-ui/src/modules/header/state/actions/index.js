import { LOGIN, LOGOUT, FETCH_USER } from "./types";
import axiosWrapper from "../../../../apis/axiosCreate";
export const login = object => {
  return {
    type: LOGIN,
    payload: object
  };
};

export const logout = () => {
  return {
    type: LOGOUT,
    payload: null
  };
};

export const createUser = userDetails => async dispatch => {
  await axiosWrapper.post(`/users`, userDetails);
  return true;
};
