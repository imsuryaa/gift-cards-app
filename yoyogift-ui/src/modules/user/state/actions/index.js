import { SENT_CARDS, RECEIVED_CARDS, USER_DETAILS, REDEEM_CARD } from "./types";
import axiosWrapper from '../../../../apis/axiosCreate';

export const fetchReceivedCards = (email) => async (dispatch) => {
    
    const response = await axiosWrapper.get(`/giftTransact?receiverEmail=${email}`);
    
    dispatch({
        type: RECEIVED_CARDS,
        payload: response.data
    })
}

export const fetchSentCards = (email) => async (dispatch) => {
    const response = await axiosWrapper.get(`/giftTransact?senderEmail=${email}`);
    dispatch({
        type: SENT_CARDS,
        payload: response.data
    })
}

export const userDetails = (id) => async (dispatch) => {    
    const response = await axiosWrapper.get(`/users/${id}`);
    dispatch({
        type: USER_DETAILS,
        payload: response.data
    })
}

export const redeemCard = (rowId, addObj) => async (dispatch) => {
    
    await axiosWrapper.delete(`/giftTransact/${rowId}`);
    const endRes = await axiosWrapper.post(`/giftTransact`, addObj);

    dispatch({
        type: REDEEM_CARD,
        payload: endRes.data
    })
    
}

export const updateUserBalance = (id, newBalance) => async () => {
    await axiosWrapper.patch(`/users/${id}`, {balance_points: newBalance})
    .catch((err) => {
        console.log(err)
    })
}

export const updateTransact = (TransactObj) => async () => {
    await axiosWrapper.post('/giftTransact', TransactObj )
    .catch((err) => {
        console.log(err)
    })   
}

export const fetchFilteredSentCards = (email, rowLimit, pageNo) => async (dispatch) => {
    const response = await axiosWrapper.get(`/giftTransact?senderEmail=${email}`,{
        params: {
          _limit: rowLimit,
          _page: pageNo
         }
      });
    dispatch({
        type: SENT_CARDS,
        payload: response.data
    })
}