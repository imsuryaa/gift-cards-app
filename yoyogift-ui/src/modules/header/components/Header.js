import React, { Component } from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import { connect } from "react-redux";
import { login, logout, createUser } from "../state/actions";
import { bindActionCreators } from "redux";
import history from "../../common/components/history";
import MySnackBar from "../../common/components/Snackbar";
import Styles from '../../../assets/css/Header.module.css';

import { GoogleLogout } from 'react-google-login';
    
const styles = {
  root: {
    flexGrow: 1,
    flexShrink: 1,
    width: '100%' 
  },
  grow: {
    flexGrow: 1
  },
  toolBar: {
    background: "#32567e"
  }
};

export class Header extends Component {
  constructor(props){
    super(props)
    this.state = {
      showErrorSnack: false
    }
  }
  render() {
    const { showErrorSnack } = this.state
    const { classes } = this.props;
    return (
      <div className={classes.root}>
        {
          showErrorSnack ? 
          <MySnackBar message='Network Error! Please try again' color='red' />
          :
          null
        }
        <AppBar position="static">
          <Toolbar className={classes.toolBar}>
            <Typography variant="h6" color="inherit" className={classes.grow}>
              <Button onClick={this.goTolanding}>
                <span style={{ fontSize: "1.2em", color: "#ffffff" }}>
                  YOYOGift
                </span>
              </Button>
            </Typography>
            {/* {this.props.isLoggedIn ? <Button color="inherit" onClick={this.addUpdateForm}>ADD UPDATE FORM</Button> : null} */}
            {this.props.isLoggedIn ? (
              <Button className={Styles.headerButton} color="inherit" onClick={this.giftsReceived}>
                GIFTS RECEIVED
              </Button>
            ) : null}
            {this.props.isLoggedIn ? (
              <Button className={Styles.headerButton} color="inherit" onClick={this.giftsSend}>
                GIFTS SENT
              </Button>
            ) : null}
            {this.props.isLoggedIn ? (
              <Button className={Styles.headerButton} color="inherit" onClick={this.myProfile}>
                MY PROFILE
              </Button>
            ) : null}
            <Button className={Styles.headerButton}
              color="inherit"
              onClick={() => { this.handleLoginLogout() }}
            >
              {this.props.isLoggedIn ?
              <GoogleLogout
              clientId="228935478905-lrnl1u77c007jhtnevha0a6jl1vctsm4.apps.googleusercontent.com"
              buttonText="Logout"
              onLogoutSuccess={this.logOut}
            >
            </GoogleLogout> :
            "LOGIN"}
            </Button>
          </Toolbar>
        </AppBar>
      </div>
    );
  }

  goTolanding = () => {
    history.push("/");
  };
  myProfile = () => {
    history.push("/Profile");
  };

  giftsSend = () => {
    history.push("/GiftsSend");
  };

  giftsReceived = () => {
    history.push("/GiftsReceived");
  };

  logOut = () => {
    this.props.logout();
    history.push("/");
    window.sessionStorage.removeItem("user");
    window.sessionStorage.removeItem("usertype");
  };

  handleLoginLogout = () => {
    if(!this.props.isLoggedIn) history.push("/login");
  };
}

Header.propTypes = {
  classes: PropTypes.object.isRequired
};

const mapStateToProps = state => {
  return {
    isLoggedIn: state.login.loginStatus,
    userDetails: state.login.detailsObject
  };
};

export const mapDispatchToProps = dispatch => {
  return bindActionCreators({ login, logout, createUser }, dispatch);
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withStyles(styles)(Header));
