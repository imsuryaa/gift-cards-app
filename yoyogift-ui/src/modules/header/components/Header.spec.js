import { mount } from "enzyme";
import React from "react";
import { Header, mapDispatchToProps } from "./Header";
import Button from "@material-ui/core/Button";
import { LOGOUT } from "../state/actions/types";

describe("Header Test Suite", () => {

  it("should find 2 buttons when user is not logged in", () => {
    //Mocks
    let classes = {root: {marginTop: "1px"}};

    let wrapper = mount(<Header classes={classes}
                        isLoggedIn={false} />);

    expect(wrapper.find("button").length).toBe(2);
    expect(wrapper.find('button').at(1).text()).toBe('LOGIN');
  });

  it("should find 6 buttons when user is logged in", () => {
    //Mocks
    let classes = {root: {marginTop: "1px"}};

    let wrapper = mount(<Header classes={classes}
                        isLoggedIn={true} />);

    expect(wrapper.find("button").length).toBe(6);
    expect(wrapper.find('button').at(1).text()).toBe('GIFTS RECEIVED');
    expect(wrapper.find('button').at(2).text()).toBe('GIFTS SENT');
    expect(wrapper.find('button').at(3).text()).toBe('MY PROFILE');
    expect(wrapper.find('button').at(4).text()).toBe('Logout');
  });

  it('should call landing function', () => {
    //Mocks
    let classes = {root: {marginTop: "1px"}};

    const wrapper = mount(<Header 
      classes={classes}
      isLoggedIn={true}
      />)
    const component = wrapper.instance();
    const spy =  jest.spyOn(component, 'goTolanding');
    
    wrapper.find(Button)
    .at(0)
    .props()
    .onClick();

    component.goTolanding();
   
   expect(component.goTolanding).toHaveBeenCalled();
   expect(component.goTolanding).toHaveBeenCalledTimes(1)
  });

  it('should call gift received function', () => {
    //Mocks
    let classes = {root: {marginTop: "1px"}};

    const wrapper = mount(<Header 
      classes={classes}
      isLoggedIn={true}
      />)
    const component = wrapper.instance();
    const spy =  jest.spyOn(component, 'giftsReceived');
    
    wrapper.find(Button)
    .at(1)
    .props()
    .onClick();

    component.giftsReceived();
   
   expect(component.giftsReceived).toHaveBeenCalled();
   expect(component.giftsReceived).toHaveBeenCalledTimes(1)
  });

  it('should call gift send function', () => {
    //Mocks
    let classes = {root: {marginTop: "1px"}};

    const wrapper = mount(<Header 
      classes={classes}
      isLoggedIn={true}
      />)
    const component = wrapper.instance();
    const spy =  jest.spyOn(component, 'giftsSend');
    
    wrapper.find(Button)
    .at(2)
    .props()
    .onClick();

    component.giftsSend();
   
   expect(component.giftsSend).toHaveBeenCalled();
   expect(component.giftsSend).toHaveBeenCalledTimes(1)
  });

  it('should call my profile function', () => {
    //Mocks
    let classes = {root: {marginTop: "1px"}};

    const wrapper = mount(<Header 
      classes={classes}
      isLoggedIn={true}
      />)
    const component = wrapper.instance();
    const spy =  jest.spyOn(component, 'myProfile');
    
    wrapper.find(Button)
    .at(3)
    .props()
    .onClick();

    component.myProfile();
   
   expect(component.myProfile).toHaveBeenCalled();
   expect(component.myProfile).toHaveBeenCalledTimes(1)
  });

  it('should call login logout function', () => {
    //Mocks
    let classes = {root: {marginTop: "1px"}};

    const wrapper = mount(<Header 
      classes={classes}
      isLoggedIn={true}
      />)
    const component = wrapper.instance();
    const spy =  jest.spyOn(component, 'handleLoginLogout');
    
    wrapper.find(Button)
    .at(4)
    .props()
    .onClick();
   
   expect(component.handleLoginLogout).toHaveBeenCalled();
   expect(component.handleLoginLogout).toHaveBeenCalledTimes(1)
  });

  it("should call log out function", () => {
    //Mocks
    let classes = {root: {marginTop: "1px"}};
    let logout= jest.fn()

    let wrapper = mount(<Header 
                          classes={classes}
                          isLoggedIn={true}
                          logout={logout}
                        />);
    
    wrapper.instance().logOut();

    expect(logout).toHaveBeenCalled();
    expect(logout).toHaveBeenCalledTimes(1);
  });

  it('should render the expected HTML when user is logged out', () => {
    //Mocks
    let classes = {root: {marginTop: "1px"}};
    expect(mount(<Header 
      classes={classes}
      isLoggedIn={false}/>
    ).html()).toMatchSnapshot();
  });

  it('should render the expected HTML when user is logged in', () => {
    //Mocks
    let classes = {root: {marginTop: "1px"}};
    expect(mount(<Header 
      classes={classes}
      isLoggedIn={true}/>
    ).html()).toMatchSnapshot();
  });

  it("should log out user", () => {
    let dispatch = jest.fn();

    let props = mapDispatchToProps(dispatch);
    props.logout();

    expect(dispatch).toHaveBeenCalled();
    expect(dispatch).toHaveBeenCalledTimes(1);

    expect(dispatch.mock.calls[0]).toEqual([{
      type: LOGOUT,
      payload: null
    }]);
  });
});