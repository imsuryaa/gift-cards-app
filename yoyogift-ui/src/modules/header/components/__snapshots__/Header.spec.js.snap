// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Test Suite should render the expected HTML when user is logged in 1`] = `"<div class=\\"[object Object]\\"><header class=\\"MuiPaper-root-10 MuiPaper-elevation4-16 MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorPrimary-8\\"><div class=\\"MuiToolbar-root-37 MuiToolbar-regular-39 MuiToolbar-gutters-38\\"><h6 class=\\"MuiTypography-root-41 MuiTypography-h6-58 MuiTypography-colorInherit-70\\"><button class=\\"MuiButtonBase-root-103 MuiButton-root-77 MuiButton-text-79 MuiButton-flat-82\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label-78\\"><span style=\\"font-size: 1.2em; color: rgb(255, 255, 255);\\">YOYOGift</span></span><span class=\\"MuiTouchRipple-root-106\\"></span></button></h6><button class=\\"MuiButtonBase-root-103 MuiButton-root-77 MuiButton-text-79 MuiButton-flat-82 MuiButton-colorInherit-98 headerButton\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label-78\\">GIFTS RECEIVED</span><span class=\\"MuiTouchRipple-root-106\\"></span></button><button class=\\"MuiButtonBase-root-103 MuiButton-root-77 MuiButton-text-79 MuiButton-flat-82 MuiButton-colorInherit-98 headerButton\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label-78\\">GIFTS SENT</span><span class=\\"MuiTouchRipple-root-106\\"></span></button><button class=\\"MuiButtonBase-root-103 MuiButton-root-77 MuiButton-text-79 MuiButton-flat-82 MuiButton-colorInherit-98 headerButton\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label-78\\">MY PROFILE</span><span class=\\"MuiTouchRipple-root-106\\"></span></button><button class=\\"MuiButtonBase-root-103 MuiButton-root-77 MuiButton-text-79 MuiButton-flat-82 MuiButton-colorInherit-98 headerButton\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label-78\\"><button style=\\"background-color: rgb(255, 255, 255); display: inline-flex; align-items: center; color: rgba(0, 0, 0, 0.54); box-shadow: 0 2px 2px 0 rgba(0, 0, 0, .24), 0 0 1px 0 rgba(0, 0, 0, .24); padding: 0px; border-radius: 2px; border: 1px solid transparent; font-size: 14px; font-weight: 500; font-family: Roboto, sans-serif; opacity: 0.6;\\" type=\\"button\\" disabled=\\"\\"><div style=\\"margin-right: 10px; background: rgb(255, 255, 255); padding: 10px; border-radius: 2px;\\"><svg width=\\"18\\" height=\\"18\\" xmlns=\\"http://www.w3.org/2000/svg\\"><g fill=\\"#000\\" fill-rule=\\"evenodd\\"><path d=\\"M9 3.48c1.69 0 2.83.73 3.48 1.34l2.54-2.48C13.46.89 11.43 0 9 0 5.48 0 2.44 2.02.96 4.96l2.91 2.26C4.6 5.05 6.62 3.48 9 3.48z\\" fill=\\"#EA4335\\"></path><path d=\\"M17.64 9.2c0-.74-.06-1.28-.19-1.84H9v3.34h4.96c-.1.83-.64 2.08-1.84 2.92l2.84 2.2c1.7-1.57 2.68-3.88 2.68-6.62z\\" fill=\\"#4285F4\\"></path><path d=\\"M3.88 10.78A5.54 5.54 0 0 1 3.58 9c0-.62.11-1.22.29-1.78L.96 4.96A9.008 9.008 0 0 0 0 9c0 1.45.35 2.82.96 4.04l2.92-2.26z\\" fill=\\"#FBBC05\\"></path><path d=\\"M9 18c2.43 0 4.47-.8 5.96-2.18l-2.84-2.2c-.76.53-1.78.9-3.12.9-2.38 0-4.4-1.57-5.12-3.74L.97 13.04C2.45 15.98 5.48 18 9 18z\\" fill=\\"#34A853\\"></path><path fill=\\"none\\" d=\\"M0 0h18v18H0z\\"></path></g></svg></div><span style=\\"font-weight: 500; padding: 10px 10px 10px 0px;\\">Logout</span></button></span><span class=\\"MuiTouchRipple-root-106\\"></span></button></div></header></div>"`;

exports[`Header Test Suite should render the expected HTML when user is logged out 1`] = `"<div class=\\"[object Object]\\"><header class=\\"MuiPaper-root-10 MuiPaper-elevation4-16 MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorPrimary-8\\"><div class=\\"MuiToolbar-root-37 MuiToolbar-regular-39 MuiToolbar-gutters-38\\"><h6 class=\\"MuiTypography-root-41 MuiTypography-h6-58 MuiTypography-colorInherit-70\\"><button class=\\"MuiButtonBase-root-103 MuiButton-root-77 MuiButton-text-79 MuiButton-flat-82\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label-78\\"><span style=\\"font-size: 1.2em; color: rgb(255, 255, 255);\\">YOYOGift</span></span><span class=\\"MuiTouchRipple-root-106\\"></span></button></h6><button class=\\"MuiButtonBase-root-103 MuiButton-root-77 MuiButton-text-79 MuiButton-flat-82 MuiButton-colorInherit-98 headerButton\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label-78\\">LOGIN</span><span class=\\"MuiTouchRipple-root-106\\"></span></button></div></header></div>"`;
